packages/cli/src/databases/repositories/annotationTagMapping.repository.ts

```diff
@@ -0,0 +1,39 @@
+import { Service } from 'typedi';
+import { DataSource, Repository } from '@n8n/typeorm';
+import { AnnotationTagMapping } from '@db/entities/AnnotationTagMapping';
+import { ExecutionAnnotation } from '@db/entities/ExecutionAnnotation';
+import { ApplicationError } from 'n8n-workflow';
+
+@Service()
+export class AnnotationTagMappingRepository extends Repository<AnnotationTagMapping> {
+	constructor(dataSource: DataSource) {
+		super(AnnotationTagMapping, dataSource.manager);
+	}
+
+	async overwriteTags(executionId: string, tagIds: string[]) {
+		return await this.manager.transaction(async (tx) => {
+			const annotation = await tx.findOne(ExecutionAnnotation, {
+				where: { execution: { id: executionId } },
+			});
+
+			if (!annotation) {
+				throw new ApplicationError(`Annotation for execution ${executionId} not found`);
```

Comment:
This method should not be called if the annotation does not exist. Probably I'll just refactor this to accept annotationId instead of executionId to make this explicit.